docs/make.jl

```diff
@@ -56,8 +56,11 @@ makedocs(
 
 include("clean_build_folder.jl")
 
-deploydocs(
-    repo = "github.com/CliMA/ClimateMachine.jl.git",
-    target = "build",
-    push_preview = true,
-)
+
+if get(ENV, "CLIMATEMACHINE_CI_DEPLOY_DOCS", "") == "true"
```

Comment:
Is this necessary? I figured that there is some internal check. Also, there's a [deploy_config](https://juliadocs.github.io/Documenter.jl/stable/lib/public/#Documenter.deploydocs) kwarg. I think our GHA docs check for `ENV["DOCUMENTER_KEY"]`.

Reply:
hmm no, but I do get an exception with the current default behaviour?  it looks like it's picking up `GITHUB_TOKEN` env variable for deployment.

Reply:
Strange, what’s the exception?

Reply:
I agree, this shouldn't be necessary. It also breaks the docs preview.